src/client.ts

```diff
@@ -80,6 +81,7 @@ export interface ClientOptions {
   authToken?: string;
   jwt?: string;
   jid?: string;
+  jidResource?: string;
```

Comment:
Would it be too much to add code comments on these options? I find that really helpful for the code hinting and if I need to `cmd + click` the definition to take me to the `*.d.ts` file. 

Also wouldn't hurt to type out the `config` for the `Client` class.

Reply:
Someday we'll have to change the doc to match whatever you are doing with the sdk.